File: src/generation/typescript/emitter/features/TupleArray.ts
function serializeTupleArray<T>(
    value: T[],
    serializer: Serializer,
    serializeElement: (item: T, serializer: Serializer) => void,
): void {
    value.forEach((item) => {
        serializeElement(item, serializer);
    });
}

function deserializeTupleArray<T>(
    deserializer: Deserializer,
    size: number,
    deserializeElement: (deserializer: Deserializer) => T,
): T[] {
    const list: T[] = [];
    for (let i = 0; i < size; i++) {
        list.push(deserializeElement(deserializer));
    }
    return list;
}

File: crates/facet-generate/src/generation/swift/emitter/features/TupleArray.swift
func serializeTupleArray<T, S: Serializer>(
    value: [T],
    serializer: S,
    serializeElement: (T, S) throws -> Void
) throws {
    for item in value {
        try serializeElement(item, serializer)
    }
}

func deserializeTupleArray<T, D: Deserializer>(
    deserializer: D,
    size: Int,
    deserializeElement: (D) throws -> T
) throws -> [T] {
    var obj: [T] = []
    for _ in 0..<size {
        obj.append(try deserializeElement(deserializer))
    }
    return obj
}
